docs/apache-airflow/concepts/dags.rst

```diff
@@ -269,11 +269,14 @@ By default, a DAG will only run a Task when all the Tasks it depends on are succ
 Branching
 ~~~~~~~~~
 
-You can make use of branching in order to tell the DAG *not* to run all dependent tasks, but instead to pick and choose one or more paths to go down. This is where the branching Operators come in.
+You can make use of branching in order to tell the DAG *not* to run all dependent tasks, but instead to pick and choose one or more paths to go down. This is where the ``@task.branch`` decorator come in.
 
-The ``BranchPythonOperator`` is much like the PythonOperator except that it expects a ``python_callable`` that returns a task_id (or list of task_ids). The task_id returned is followed, and all of the other paths are skipped. It can also return None to skip all downstream task.
+The ``@task.branch`` is much like the ``@task`` decorator except that it expects a ``python_callable`` that returns a task_id (or list of task_ids). The task_id returned is followed, and all of the other paths are skipped. It can also return None to skip all downstream task.
 
-The task_id returned by the Python function has to reference a task directly downstream from the BranchPythonOperator task.
+.. warning::
+    The ``@task.branch`` decorator is recommended over the classic :class:`~airflow.operators.python.BranchPythonOperator`.
```

Comment:
This probably should be a `note`, `warning` is a bit strong.

Reply:
This note can probably be moved to under the part where we mention subclassing `BranchPythonOperator`, and say something like this instead:

> The `@task.branch` decorator is recommended over directly instantiating :class:`~airflow.operators.python.BranchPythonOperator` in a DAG.
```